### PR TITLE
Add default.nix and nixfy cli-build

### PR DESCRIPTION
**Pull request by jmininger**
Dependent on #39

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,4 +1,9 @@
-{ pkgs ? import ../../nixpkgs {} }:
+{ pkgs ? import (builtins.fetchTarball {
```
Comment (3noch) [inline default.nix:1]: Wasn't it already pinned? We should always use the *same* pin unless there's a very good reason not to.

Comment (jmininger) [reply default.nix:1]: @3noch This *is* the original pin (from the shell.nix). The removed line was pointing to nixpkgs in my home path, which had a bunch of trace statements lying around

File: nix/avalanche.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  name = "gecko";
+  src = import ./dep/gecko/thunk.nix;
+  vendorSha256 = "1z6iy389j5an2qsjfjib2hcal0fpqkj49qarcib5vlgsy7skmnzw";
+  doCheck = false;
+  buildPhase = ''
+    mkdir -p $out
+    mkdir -p $out/bin
+    mkdir -p $out/plugins
+    cp ${coreth}/bin/evm $out/plugins/evm
+    go build -mod=vendor -o $out/bin/avalanche "$src/main/"*.go
```
Comment (tomsmalley) [inline nix/avalanche.nix:15]: why `avalanche` instead of `gecko`?

Comment (jmininger) [reply nix/avalanche.nix:15]: @tomsmalley its as per scripts/build.sh (or one of those scripts)...Doesn't need to be though; its just how *they* do it

Comment (tomsmalley) [reply nix/avalanche.nix]: ahh, I see.

File: package.json [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
+  "name": "avalanche-wallet-cli",
   "private": true,
-  "scripts": {
-    "avalanchejs": "cd avalanche.js && npm install && npm run build",
-    "ledgerjs": "[ \"$(ls -A ledgerjs)\" ] || git submodule update --recursive --init ledgerjs; cd ledgerjs; yarn install --frozen-lockfile && yarn build",
-    "preinstall": "yarn ledgerjs && yarn avalanchejs",
-    "build": "yarn pkg cli"
-  },
+	"scripts": {
+		"hw-app-avalanche": "cd hw-app-avalanche && yarn install && yarn run build",
```
Comment (tomsmalley) [inline package.json:5]: Should this have --frozen-lockfile too?

Comment (3noch) [reply package.json:5]: Yeah it should

File: nix/avalanche.nix [hunk 4]
```diff
@@ -1,10 +1,20 @@
 { pkgs ? import <nixpkgs> {} }:
 let
   coreth = (import ./coreth.nix) {};
+	fetchThunk = p:
```
Comment (3noch) [inline nix/avalanche.nix:4]: We should use r-p for this. Also no tabs.

File: nix/coreth.nix [hunk 5]
```diff
@@ -1,9 +1,49 @@
-{ pkgs ? import <nixpkgs> {} }:
-pkgs.buildGoModule {
+{ pkgs ? import <nixpkgs>{}}:
```
Comment (3noch) [inline nix/coreth.nix:1]: Noods to be pinned?

**Pull request merged**